src/aes_key.c

```diff
@@ -652,14 +677,23 @@ zpc_aes_key_import(struct zpc_aes_key *aes_key, const unsigned char *buf,
 		goto ret;
 	}
 
-	if (!aes_key_blob_has_valid_mkvp(aes_key, buf, buflen)) {
-		rc = ZPC_ERROR_WKVPMISMATCH;
-		goto ret;
+	if (aes_key->type != ZPC_AES_KEY_TYPE_PVSECRET) {
+		if (!aes_key_blob_has_valid_mkvp(aes_key, buf, buflen)) {
+			rc = ZPC_ERROR_WKVPMISMATCH;
+			goto ret;
+		}
+
+		if (!aes_key_blob_is_pkey_extractable(aes_key, buf, buflen)) {
+			rc = ZPC_ERROR_BLOB_NOT_PKEY_EXTRACTABLE;
+			goto ret;
+		}
 	}
 
-	if (!aes_key_blob_is_pkey_extractable(aes_key, buf, buflen)) {
-		rc = ZPC_ERROR_BLOB_NOT_PKEY_EXTRACTABLE;
-		goto ret;
+	if (aes_key->type == ZPC_AES_KEY_TYPE_PVSECRET) {
```

Comment:
this could be an else of above if clause.

Reply:
Done.